picoquic/logger.c

```diff
@@ -1811,73 +1811,45 @@ void picoquic_log_picotls_ticket(FILE* F, picoquic_connection_id_t cnx_id,
 
 void picoquic_open_cc_dump(picoquic_cnx_t * cnx)
 {
-    int ret = 0;
-
     if (cnx->cc_log != NULL) {
         DBG_PRINTF("%s", "CC LOG File is already open!\n");
-    } else if (cnx->quic->cc_log_dir != NULL) {
-        char cc_log_file_name[512];
-        char const * suffix = "-log.bin";
-        const size_t suffix_len = strlen(suffix);
-        size_t folder_length = strlen(cnx->quic->cc_log_dir);
-
-        if (folder_length + 1u + (size_t)cnx->initial_cnxid.id_len*2u + suffix_len + 1u > sizeof(cc_log_file_name)) {
-            ret = -1;
-        }
-        else {
-            size_t byte_index = 0;
-
-            memcpy(cc_log_file_name, cnx->quic->cc_log_dir, folder_length);
-            byte_index += folder_length;
-#ifdef _WINDOWS
-            cc_log_file_name[byte_index++] = '\\';
-#else
-            cc_log_file_name[byte_index++] = '/';
-#endif
-            for (size_t i = 0; ret == 0 && i < cnx->initial_cnxid.id_len; i++) {
-#ifdef _WINDOWS
-                ret = sprintf_s(&cc_log_file_name[byte_index], sizeof(cc_log_file_name) - byte_index, "%02x", cnx->initial_cnxid.id[i]) <= 0;
-#else
-                ret = sprintf(&cc_log_file_name[byte_index], "%02x", cnx->initial_cnxid.id[i]) <= 0;
-#endif
-
-                byte_index += 2;
-            }
+        return;
+    }
 
-            if (ret == 0) {
-                memcpy(&cc_log_file_name[byte_index], suffix, suffix_len);
-                byte_index += suffix_len;
-                cc_log_file_name[byte_index] = 0;
-            }
+    if (cnx->quic->cc_log_dir == NULL) {
+        DBG_PRINTF("%s", "CC LOG directory not set!\n");
```

Comment:
This trace comes now pretty often during unit tests. I'll fix that in a later PR.

Reply:
The API is designed  for requesting logs on the server side. That cannot be done on a per connection basis. Setting the cc_log directory signals that the server should create one file per connection in that directory.

Reply:
The code still does the same thing it did before. This is meant to be a refactoring only without changing any behavior.

The library still creates a log file per connection when the cc_log_dir is set to != NULL. My comment was just a reminder that the unittests, that don't set a directory, now complain a lot. I fixed that on another branch already.